### DISCUSSION TITLE
Why package with async

### DISCUSSION THREAD

**Discussion by gbpoole** [sindresorhus/pure#637] [Q&A]
Just curious: why do you package async.zsh with Pure, when it's separately available (and can be kept up-to-date) as a plugin?

**Comment by mafredri** [comment]
It's mostly for convenience and ease of installation as not all users have a plugin manager. This way a user can just check out or download this repository and add it to their `fpath`. Pure is also available on npm whereas zsh-async is not, so to make that installation path smooth, it needs to be bundled one way or another. Perhaps the final motivation is to bundle a known working version of `zsh-async` but that has the downside of the bundled `zsh-async` sometimes lagging behind.

That said, it's totally fine to manage Pure and zsh-async as separate plugins, you can pick `pure.zsh` / `prompt_pure_setup` from Pure and `async.zsh` from `zsh-async`.

**Comment by gbpoole** [comment]
Terrific.  Thanks for that clear and prompt answer!
